examples/cms/src/FieldMapping.tsx

```diff
@@ -0,0 +1,244 @@
+import type { DataSource, syncCollection } from "./data"
+
+import { ManagedCollection, ManagedCollectionField, framer } from "framer-plugin"
+import { useState, useMemo, useLayoutEffect } from "react"
+import { computeFieldsFromDataSource, mergeFieldsWithExistingFields } from "./data"
+import { Spinner } from "./components/Spinner"
+import { UI_DEFAULTS } from "./constants"
+
+interface FieldMappingRowProps {
+    originalField: ManagedCollectionField
+    field: ManagedCollectionField
+    isIgnored: boolean
+    onFieldToggle: (fieldId: string) => void
+    onFieldNameChange: (fieldId: string, name: string) => void
+}
+
+function FieldMappingRow({ originalField, field, isIgnored, onFieldToggle, onFieldNameChange }: FieldMappingRowProps) {
+    const isUnsupported = !field
+    const hasFieldNameChanged = field!.name !== originalField.name
+    const fieldName = hasFieldNameChanged ? field!.name : ""
+    const placeholder = isUnsupported ? "Unsupported Field" : originalField.name
+    const isDisabled = isUnsupported || isIgnored
+
+    return (
+        <>
+            <div
+                className="source-field"
+                aria-disabled={isDisabled}
+                onClick={() => onFieldToggle(field!.id)}
+                role="button"
+            >
+                <input
+                    type="checkbox"
+                    disabled={isUnsupported}
+                    checked={!isDisabled}
+                    onChange={event => {
+                        event.stopPropagation()
+                    }}
```

Comment:
The stopPropagation() prevents the click event from reaching the parent div, which would otherwise trigger - once from the checkbox and once from the div's click handler.

Reply:
How is the checkbox change handled though? I don't see a change handler

Reply:
What I mean is that the checkbox input should be focusable by keyboard, but then using the keyboard to check or uncheck it would not do anything I guess, right? If so, that's bad accessibility

Reply:
Yeah, actually it works! I added a box shadow to outline the whole input when using the keyboard 🦾

Reply:
How does it work when you are not listening for events?

Reply:
When you interact with the checkbox (either by clicking or using keyboard Space/Enter):

- The browser fires a click event
- Our `onChange` handler only does stopPropagation() to prevent double-triggering
- The click event bubbles up to the parent div's handler which toggles the field (and updates isIgnored)
- The checkbox reflects this through `checked={!isIgnored}`

So while it might look like we're not handling the checkbox events, we are - just at the parent div level rather than directly on the checkbox. This pattern works well here since we want the entire row to be clickable anyway.

Reply:
TIL. I had no idea it worked like this. Very strange still

Reply:
Btw. I just checked in a code sandbox and I don't think the `stopPropagation` is needed. I tried both clicking the checkbox and outer div, and both resulted in a single click event.